File: src/app/shared/service/api/securityPolicy.service.ts
import { HttpClient, HttpRequest } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { SecurityPolicyDTORequest } from '../../model/request/securityPolicyDTORequest';

@Injectable({
    providedIn: 'root'
})
export class SecurityPolicyService {
    constructor(
        private _httpClient: HttpClient
    ) { }

    public findByCode(): Observable<any> {
        return this._httpClient.get<any>(environment.urlBase + '/securities-policies/security-policy');
    }

    public update(dto: SecurityPolicyDTORequest, id: number) {
        const requ = new HttpRequest('PUT', `${environment.urlBase}/securities-policies/security-policy/` + id, dto, {
            reportProgress: true,
            responseType: 'json'
        });
        return this._httpClient.request(requ);
    }

}

File: src/main/java/com/sys/botica/crce/pe/sys_botica/service/SecurityPolicyService.java
package com.sys.botica.crce.pe.sys_botica.service;

import com.sys.botica.crce.pe.sys_botica.dto.HrefEntityDTO;
import com.sys.botica.crce.pe.sys_botica.dto.SecurityPolicyDTO;
import com.sys.botica.crce.pe.sys_botica.dto.request.SecurityPolicyDTORequest;

public interface SecurityPolicyService {
	public HrefEntityDTO update(SecurityPolicyDTORequest dto,Long id);
	public SecurityPolicyDTO findByCode();
}
